### PR TITLE
Smart wallet demo e2e tests

### PR DESCRIPTION
**Pull request by jmderby**
## Description

This PR covers the React Smart Wallet end-to-end (E2E) tests written with Playwright. The objective is to ensure 100% confidence that our Smart Wallet SDK won't break when making changes during local development. Additionally, a test script has been added to run the E2E tests against the local SDK for increased confidence.

## Test plan

- Ensure packages and tests are easy to execute from the SDK using the following command (from the root sdk directory):
`pnpm run test:starter` to run the e2e tests
- Ensure tests cover the critical paths of the Smart Wallet demo.

### CODE REVIEW

File: .changeset/thirty-dancers-bake.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@crossmint/client-sdk-verifiable-credentials": minor
+---
+
+Added crossmint encryption
```
Comment (jmderby) [inline .changeset/thirty-dancers-bake.md:5]: this file can be removed, not sure why it carried over :thinking:

File: apps/wallets/smart-wallet/react/.github/workflows/playwright.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+name: Playwright Tests
```
Comment (jmderby) [inline apps/wallets/smart-wallet/react/.github/workflows/playwright.yml:1]: May require some tweaking in order to get CI working as expected.

File: apps/wallets/smart-wallet/react/.github/workflows/playwright.yml
Comment (devlyn37) [inline apps/wallets/smart-wallet/react/.github/workflows/playwright.yml:1]: I'd leave this for another PR if you're not confident it works yet!

File: apps/wallets/smart-wallet/react/src/app/utils/createAAWallet/createViemWallet.ts [hunk 3]
```diff
@@ -1,10 +1,16 @@
...
 
-import { checkAuthState, parseToken, signInWithGoogle } from "../../auth/FirebaseAuthManager";
+import { checkAuthState, signInWithGoogle } from "../../auth/FirebaseAuthManager";
+
+export const createViemAAWallet = async (isProd: boolean, privateKey: `0x${string}`) => {
+    if (!privateKey) {
+        throw new Error(
+            "No private key found: Please provide a private key to this function or use the createPasskeyWallet function"
+        );
+    }
```
Comment (devlyn37) [inline apps/wallets/smart-wallet/react/src/app/utils/createAAWallet/createViemWallet.ts:12]: If we need to check this is the type above correct?

File: apps/wallets/smart-wallet/react/src/app/utils/createAAWallet/createViemWallet.ts [hunk 4]
```diff
@@ -25,26 +29,12 @@ export const createViemAAWallet = async (isProd: boolean) => {
...
-
-    const account = mnemonicToAccount(mnemonic) as any;
     // NOTE: Do NOT do this in production. This is just for demo purposes.
     // Proper storage of private key material is critical.
     // Crossmint supports several secure signer options, documented later in the guide.
-
     const signer: ViemAccount = {
         type: "VIEM_ACCOUNT",
-        account,
+        account: privateKeyToAccount(privateKey) as any,
```
Comment (devlyn37) [inline apps/wallets/smart-wallet/react/src/app/utils/createAAWallet/createViemWallet.ts:37]: Maybe worth spending a few minutes to see if there's a more specific casting we can do.

File: .changeset/thirty-dancers-bake.md (hunk 1 above)
Comment (devlyn37) [reply .changeset/thirty-dancers-bake.md:5]: Can you do so in your branch?

File: apps/wallets/smart-wallet/react/src/app/utils/createAAWallet/createViemWallet.ts (hunk 4 above)
Comment (devlyn37) [reply apps/wallets/smart-wallet/react/src/app/utils/createAAWallet/createViemWallet.ts:37]: I wonder if updating the viem package to the same one the sdk uses would help here?

File: apps/wallets/smart-wallet/react/README.md [hunk 5]
```diff
@@ -52,3 +57,51 @@ pnpm run dev
 ```
 
 This command will launch the application on `http://localhost:3000` by default, allowing you to view and interact with it in your web browser.
+
+## Playwright setup
+
+Authenticating with Google auth requires adding this file to the project.
+
+add `playwright/.auth/user-session.json` to the root of this app.
```
Comment (devlyn37) [inline apps/wallets/smart-wallet/react/README.md:65]: Can you generate this programmatically?

File: apps/wallets/smart-wallet/react/README.md [hunk 6]
```diff
@@ -52,3 +57,51 @@ pnpm run dev
...
+pnpm exec playwright test --debug
+
+```
+
+Auto generate tests with Codegen.
+
+```bash
+pnpm exec playwright codegen
+
+```
```
Comment (devlyn37) [inline apps/wallets/smart-wallet/react/README.md:107]: All this might be overkill here

File: apps/wallets/smart-wallet/react/package.json [hunk 7]
```diff
@@ -45,6 +45,7 @@
         "@testing-library/jest-dom": "^5.16.5",
         "@types/react": "^18.2.79",
         "classnames": "^2.3.2",
+        "dotenv": "16.4.5",
```
Comment (devlyn37) [inline apps/wallets/smart-wallet/react/package.json:48]: nit: can this be a dev dep?

**Review by devlyn37 — COMMENTED** [review]
Left some quick initial comments before I check out the branch and start running stuff

File: packages/client/wallets/smart-wallet/package.json [hunk 8]
```diff
@@ -23,8 +23,10 @@
         "build-no-minify": "tsup src/index.ts --clean --format esm,cjs --outDir ./dist --dts --sourcemap",
         "dev": "tsup src/index.ts --clean --format esm,cjs --outDir ./dist --dts --sourcemap --watch",
         "link:common-sdk-base": "yalc add @crossmint/common-sdk-base && yalc link @crossmint/common-sdk-base && pnpm i && pnpm build && yalc push",
+        "link:starter": "pnpm link --global @crossmint/client-sdk-smart-wallet && pnpm --filter @crossmint/client-sdk-smart-wallet-react-starter exec pnpm link --global @crossmint/client-sdk-smart-wallet",
```
Comment (devlyn37) [inline packages/client/wallets/smart-wallet/package.json:26]: ty, much needed!

Comment (devlyn37) [reply packages/client/wallets/smart-wallet/package.json:26]: Might be worth spending 5 mins investigating if you should use yalc here

Comment (jmderby) [reply packages/client/wallets/smart-wallet/package.json:26]: np :smile:

File: apps/wallets/smart-wallet/react/README.md (hunk 5 above)
Comment (jmderby) [reply apps/wallets/smart-wallet/react/README.md:65]: turns out it already does... :sweat_smile: remooving this in readme

File: apps/wallets/smart-wallet/react/.github/workflows/playwright.yml
Comment (jmderby) [reply apps/wallets/smart-wallet/react/.github/workflows/playwright.yml:1]: I removed the test script therefore, the ci stuff wont run now

Comment (devlyn37) [reply apps/wallets/smart-wallet/react/.github/workflows/playwright.yml:1]: If it's not running I think still best to remove and keep on your machine until it works

File: packages/client/wallets/smart-wallet/package.json [hunk 9]
```diff
@@ -36,7 +38,7 @@
         "@zerodev/webauthn-key": "5.3.1",
         "permissionless": "0.1.36",
         "uuid": "10.0.0",
-        "viem": "2.17.5"
+        "viem": "2.9.17"
```
Comment (devlyn37) [inline packages/client/wallets/smart-wallet/package.json:41]: Hey, I meant the other way around!

You rather than downgrade viem here, you should upgrade viem in `smart-wallet/react` to check the type thing.

This change would break a few aspects of the SDK

**Review by devlyn37 — COMMENTED** [review]
.

File: packages/client/wallets/smart-wallet/package.json (hunk 9 above)
Comment (jmderby) [reply packages/client/wallets/smart-wallet/package.json:41]: my bad :sweat_smile:. Upgraded the react app but still didn't fix the type scream :disappointed:

**Comment by jmderby** [comment]
@devlyn37 One thing to note while testing: the webkit and firefox browsers are failing for whatever reason but Chromium seems to pass without flakes. 

Another thing to note is i've added the workspace tag to the smart-wallet sdk so there's no need to link the packages anymore. The only thing you need to do when testing is: 

1. `pnpm i`
2.  cd into the smart-wallet dir and run: `pnpm run test:starter`

File: apps/payments/vue/package.json [hunk 10]
```diff
@@ -10,8 +10,8 @@
         "type-check": "vue-tsc --noEmit"
     },
     "dependencies": {
-        "@crossmint/client-sdk-smart-wallet": "workspace:*",
         "@crossmint/client-sdk-base": "workspace:*",
+        "@crossmint/client-sdk-smart-wallet": "workspace:*",
         "@crossmint/client-sdk-vue-ui": "workspace:*",
         "vue": "3.3.4",
         "vue-router": "4.2.4"
```
Comment (jmderby) [inline apps/payments/vue/package.json:17]: not sure why this file changed, but it's formatted with prettier how it's supposed to be. Must've been something went wrong when merging with main.

File: apps/wallets/smart-wallet/react/README.md [hunk 11]
```diff
@@ -52,3 +57,23 @@ pnpm run dev
...
+
+```bash
+pnpm exec playwright test
+```
+
+Starts the interactive UI mode.
+
+```bash
+pnpm exec playwright test --ui
+```
```
Comment (devlyn37) [inline apps/wallets/smart-wallet/react/README.md:79]: redundant with `test:ui` as an option?

File: apps/wallets/smart-wallet/react/README.md [hunk 12]
```diff
@@ -52,3 +57,23 @@ pnpm run dev
 ```
 
 This command will launch the application on `http://localhost:3000` by default, allowing you to view and interact with it in your web browser.
+
+### Running the tests
+
+You can run the tests in ui mode by using the script:
+
+```bash
+pnpm run test:ui
```
Comment (devlyn37) [inline apps/wallets/smart-wallet/react/README.md:66]: don't need run here

File: apps/wallets/smart-wallet/react/src/app/utils/mintApi.ts [hunk 13]
```diff
@@ -1,7 +1,8 @@
 import { ethers } from "ethers";
 import { encodeFunctionData } from "viem";
 
-import { Blockchain, EVMSmartWallet } from "@crossmint/client-sdk-aa-passkeys-beta";
+import { EVMSmartWallet } from "@crossmint/client-sdk-smart-wallet";
+import { SmartWalletChain } from "@crossmint/client-sdk-smart-wallet";
```
Comment (devlyn37) [inline apps/wallets/smart-wallet/react/src/app/utils/mintApi.ts:5]: Double import

File: apps/wallets/smart-wallet/react/tests/auth.setup.ts
Comment (devlyn37) [inline apps/wallets/smart-wallet/react/tests/auth.setup.ts:1]: Will ignore for now with the context that this cycle we'll be adding our own auth product in here.

File: packages/client/wallets/smart-wallet/package.json [hunk 14]
```diff
@@ -25,6 +25,7 @@
         "link:common-sdk-base": "yalc add @crossmint/common-sdk-base && yalc link @crossmint/common-sdk-base && pnpm i && pnpm build && yalc push",
         "test": "cross-env NODE_ENV=test jest",
         "test-coverage": "cross-env NODE_ENV=test jest --coverage",
+        "test:starter": "pnpm run build && pnpm --filter @crossmint/client-sdk-smart-wallet-react-starter exec playwright test",
```
Comment (devlyn37) [inline packages/client/wallets/smart-wallet/package.json:28]: nice

**Review by devlyn37 — APPROVED** [review]
Nice
